src/book.py

```diff
@@ -179,6 +180,77 @@ def _read_coinbase(self, file_path: Path) -> None:
                     self.append_operation("Fee", utc_time, platform,
                                           eur_fee, "EUR", row, file_path)
 
+    def _read_kraken_trades(self, file_path: Path) -> None:
+        log.error(f"{file_path.name}: "
+                  f"Looks like this is a Kraken 'Trades' history, but we need the 'Ledgers' history. "
+                  f"(See: https://github.com/provinzio/CoinTaxman/wiki/Exchange:-Kraken)")
+
+    def _read_kraken_ledgers(self, file_path: Path) -> None:
+        platform = "kraken"
+        operation_mapping = {
+            "spend": "Sell",  # Sell ordered via 'Buy Crypto' button
+            "receive": "Buy",  # Buy ordered via 'Buy Crypto' button
+            "transfer": "Airdrop",
+            "reward": "StakingInterest",
+            "deposit": "Deposit",
+            "withdrawal": "Withdraw",
+        }
+
+        # Need to track state of "duplicate entries" for deposits / withdrawals:
+        #   dup_state["deposit"] == 1: Deposit is broadcast to blockchain            <-- Taxable event (is in public trade history)
+        #   dup_state["deposit"] == 0: Deposit is credited to Kraken account         <-- Skipped
+        #   dup_state["withdrawal"] == 1: Withdrawal is requested in Kraken account  <-- Skipped
+        #   dup_state["withdrawal"] == 0: Withdrawal is broadcast to blockchain      <-- Taxable event (is in public trade history)
+        dup_state, dup_skip = {"deposit": 0, "withdrawal": 0}, {"deposit": 0, "withdrawal": 1}
+        # See: https://support.kraken.com/hc/en-us/articles/360001169443-Why-there-are-duplicate-entries-for-deposits-withdrawals
+
+        with open(file_path, encoding="utf8") as f:
+            reader = csv.reader(f)
+
+            # Skip header.
+            next(reader)
+
+            for txid, refid, _utc_time, _type, subtype, aclass, _asset, _amount, _fee, balance in reader:
+                row = reader.line_num
+
+                # Skip "duplicate entries" for deposits / withdrawals
+                if _type in dup_state.keys():
+                    dup_state[_type] = (dup_state[_type] + 1) % 2
+                    if dup_state[_type] == dup_skip[_type]:
+                        continue
```

Comment:
I think that the counter logic is still too cumbersome to read.

We may lighten this up by changing the order of `dup_state[_type] = (dup_state[_type] + 1) % 2` and the if-case.
We have to change the skip values to 1, 0 for this to work.

This leaves us with 0 -> first entry, 1 -> second entry. Currently it is 1 -> first entry, 0 -> second entry, which I find weird.

Changing this, we can order the state comments above consecutivly (0, 1) instead of (1,0)

Reply:
Changed it accordingly.